fdb-record-layer-lucene/src/main/java/org/apache/lucene/index/StandardDirectoryReaderOptimization.java

```diff
@@ -88,4 +91,60 @@ protected DirectoryReader doBody(String segmentFileName) throws IOException {
             }
         }.run(commit);
     }
+
+    static class SegmentReaderOpener {
+        private final SegmentReader[] segmentReaders;
+        private final int readerIndex;
+
+        private final SegmentCommitInfo segmentCommitInfo;
+        private final int createdVersionMajor;
+        private final IOContext ioContext;
+
+        private IOException ioException;
+        private CompletableFuture<Void> future;
+
+        SegmentReaderOpener(final SegmentReader[] segmentReaders, final int readerIndex,
+                            final SegmentCommitInfo segmentCommitInfo, final int createdVersionMajor, final IOContext ioContext) {
+            this.segmentReaders = segmentReaders;
+            this.readerIndex = readerIndex;
+            this.segmentCommitInfo = segmentCommitInfo;
+            this.createdVersionMajor = createdVersionMajor;
+            this.ioContext = ioContext;
+        }
+
+        public void start(final Executor executor) {
+            try {
+                future = CompletableFuture.supplyAsync(() -> {
```

Comment:
I think this could just use `CompletableFuture.runAsync()`, since it doesn't have a result.

Reply:
Sure. Done.